app/controllers/products_controller.rb

```diff
@@ -305,35 +279,68 @@ def launch
 
   # private
 
-  def create_product_chat(product)
-    main_thread = product.discussions.create!(title: Discussion::MAIN_TITLE, user: current_user, number: 0)
-    product.update(main_thread: main_thread)
-    product.chat_rooms.create!(wip: main_thread, slug: product.slug)
+  def setup_core_team(product)
+    core_team_ids = Array(params[:core_team])
+    core_team_members = User.where(id: core_team_ids.select(&:uuid?))
+    core_team_members.each do |user|
+      product.core_team_memberships.create(user: user)
+    end
   end
 
-  def create_product_with_params
+  def find_the_chosen
+    chosen_ids = params[:product][:partner_ids] || ''
+    chosen_ids = chosen_ids.split(',').flatten
+  end
+
+  def disperse_coins(product, chosen_ids)
+    GiveCoinsToParticipants.new.perform(chosen_ids, product.id)
+  end
+
+  def spread_the_word(idea, product, chosen_ids)
+    if idea
+      the_key = idea.slug.to_sym
+      chosen_ids.each do |chosen_id|
+        EmailLog.send_once(chosen_id, the_key) do
+          PartnershipMailer.delay(queue: 'mailer').create(chosen_id, product.id, idea.id)
+        end
+      end
+      Tweeter.tweet_new_product(idea, product)
+    end
+  end
+
+  def create_product_with_params(idea=nil)
```

Comment:
Assignment Branch Condition size for create_product_with_params is too high. [34.26/15]<br>Surrounding space missing in default value assignment.